SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -382,4 +381,24 @@ QueryConstraint InterestFactory::CreateLevelConstraints() const
 	return LevelConstraint;
 }
 
+TArray<uint32> InterestFactory::CreateClientResultType() const
+{
+	TArray<uint32> ResultType;
+
+	// Add the required unreal components
+	ResultType.Add(SpatialConstants::UNREAL_METADATA_COMPONENT_ID);
+	ResultType.Add(SpatialConstants::SPAWN_DATA_COMPONENT_ID);
+	ResultType.Add(SpatialConstants::NETMULTICAST_RPCS_COMPONENT_ID_LEGACY);
+	ResultType.Add(SpatialConstants::MULTICAST_RPCS_COMPONENT_ID);
+	ResultType.Add(SpatialConstants::RPCS_ON_ENTITY_CREATION_ID);
+
+	// TODO: breaks without this, but shouldn't need it. Figure out why
+	ResultType.Add(SpatialConstants::ENTITY_ACL_COMPONENT_ID); 
+
+	// Add all data components- clients don't need to see handover or owner only components on other entities.
+	ResultType.Append(ClassInfoManager->SchemaDatabase->DataComponentIds.Array());
```

Comment:
I think for a bit more encapsulation, add a function on the class info manager to do this?

Reply:
also I would do `ResultType.Append(MoveTemp(DataComponentIdsArray))` That's a fat array you don't want to be copying.

Reply:
done